### ISSUE TITLE
<MessageBoxFunctionalLayout/> - Missing footer border

### ISSUE THREAD

**Issue by simast** [wix/wix-style-react#4942]
# 🐛 Bug Report

### 🏗 Relevant Components

`<MessageBoxFunctionalLayout/>`

### 😯 Current Behavior

`<MessageBoxFunctionalLayout/>` is missing a footer border when the content is updated/rerenders.

### 🤔 Expected Behavior

`<MessageBoxFunctionalLayout/>` should detect changes to content height and show footer border when required.

### 👣 Steps to Reproduce

Use this example in playground:

```jsx
() => {
  const [loading, setLoading] = React.useState(true);
  React.useEffect(() => {
    setTimeout(() => setLoading(false), 1000);
  });
  return (
    <MessageBoxFunctionalLayout
      title="Choose Your Favorites"
      confirmText="Select"
      maxHeight="200px"
      noBodyPadding
    >
      {loading ? 'loading' : (
        <Table
          data={[
            { firstName: 'Meghan', lastName: 'Bishop' },
            { firstName: 'Sara', lastName: 'Porter' },
            { firstName: 'Deborah', lastName: 'Rhodes' },
            { firstName: 'Walter', lastName: 'Jenning' },
          ]}
          columns={[
            { title: 'First', render: row => row.firstName },
            { title: 'Last', render: row => row.lastName },
          ]}
        >
          <Table.Content />
        </Table>
      )}
    </MessageBoxFunctionalLayout>
  )
};
```

After the table is rendered (delayed to 1 second) - the modal will not have a footer border separating actions. The issue is this line:

https://github.com/wix/wix-style-react/blob/d091038dd96e1f24c7f182951b8400ad21aa8d04/src/MessageBox/FunctionalLayout/MessageBoxFunctionalLayout.js#L32

We are only checking for scroll during initial mounting (when the ref is initialised).

  - **WSR version**: `7.38.1`
  - **Yoshi version**: `4.29.0`

### 👀 Severity

Low

**Comment by shlomitc** [comment]
any suggestion hos to solve this?

**Comment by simast** [comment]
Since we are already working with refs in this component - I would use `ResizeObserver` here to solve this issue. Pseudo code:

```js
class MessageBoxFunctionalLayout extends WixComponent {
  constructor(props) {
    super(props);
    // ...
    this.resizeObserver = new ResizeObserver(this._observeContentResize);
  }

  componentDidMount() {
    this.resizeObserver.observe(this.messageBoxRef);
  }

  componentWillUnmount() {
    this.resizeObserver.unobserve(this.messageBoxRef);
  }

  _observeContentResize = () => {
    // just run the same logic as currently in `_initializeMessageBoxRef`
  };
}
```

**Comment by EgoziE** [comment]
deprecated
